app/models/settings.rb

```diff
@@ -0,0 +1,33 @@
+# == Schema Information
+#
+# Table name: settings
+#
+#  id                  :integer          not null, primary key
+#  created_at          :datetime
+#  updated_at          :datetime
+#  singleton_guard     :integer
+#  current_semester_id :integer
+#  npo_app_open        :boolean
+#  student_app_open    :boolean
+#
+
+class Settings < ActiveRecord::Base
+  validates_numericality_of :singleton_guard, equal: 0
+  validates :npo_app_open, inclusion: { in: [true, false] }
+  validates :student_app_open, inclusion: { in: [true, false] }
+  validates :current_semester_id, presence: true
+
+  # source: http://stackoverflow.com/questions/399447/how-to-implement-a-singleton-model
+  def self.instance
+    # there will be only one row, and its ID must be '1'
+    begin
+      first
```

Comment:
Actually, `first` will never throw an error (it just returns `nil`), so we can either not worry about it and take out the `rescue`, we can just give up and throw an error if we don't have a Settings like:

    first || fail "No settings!"

 or we can do something more complex where we create a Settings. I'm kind of leaning towards the second option since we should definitely have initialized a Settings model for the site to function.